src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/common.targets

```diff
@@ -28,31 +26,38 @@
       <ExecArgs
         Condition="'$(MvcRazorEmbedViewSources)'=='true'"
         Include="--embed-view-sources" />
+
+      <ExecArgs Include="--file=%(MvcRazorFilesToCompile.FullPath)" />
     </ItemGroup>
 
     <ItemGroup Condition="'$(SignAssembly)'=='true'">
-      <ExecArgs 
+      <ExecArgs
         Condition="'$(DelaySign)'=='true'"
         Include="--delay-sign" />
-      <ExecArgs 
+      <ExecArgs
         Condition="'$(PublicSign)'=='true'"
         Include="--public-sign" />
       <ExecArgs Include="--key-file=$(AssemblyOriginatorKeyFile)" />
     </ItemGroup>
 
     <WriteLinesToFile
       File="$(_MvcRazorResponseFilePath)"
-      Lines="@(ExecArgs)" 
+      Lines="@(ExecArgs)"
       Overwrite="true"  />
   </Target>
 
-   <Target 
-      Name="_MvcRazorPrecompileOnPublish"
-      DependsOnTargets="MvcRazorPrecompile"
-      AfterTargets="PrepareForPublish"
-      Condition="'$(MvcRazorCompileOnPublish)'=='true'">
+  <Target
+     Name="_MvcRazorPrecompileOnPublish"
+     DependsOnTargets="MvcRazorPrecompile"
+     AfterTargets="PrepareForPublish"
+     Condition="'$(MvcRazorCompileOnPublish)'=='true'">
 
     <ItemGroup>
+      <Content
+        Update="**\*.cshtml"
```

Comment:
Good catch, I'll expand the test to cover this. The Target is already conditionally defined if we have `MvcRazorCompileOnPublish` == `true`, so we technically don't need added condition.